XUtils.c

```diff
@@ -224,6 +232,209 @@ size_t String_safeStrncpy(char* restrict dest, const char* restrict src, size_t
    return i;
 }
 
+#ifdef HAVE_LIBNCURSESW
+ATTR_NONNULL_N(3)
+static size_t String_encodeWChar(void* pos, wchar_t wc, void* state) {
+   char tempBuf[MB_LEN_MAX];
+
+   size_t len = wcrtomb((pos ? (char*)pos : tempBuf), wc, (mbstate_t*)state);
+   assert(len != (size_t)-1);
+   assert(len > 0);
+
+   return len;
+}
+#else
+static size_t String_encodeWChar(void* pos, int c, void* state) {
+   (void)state;
+
+   if (pos) {
+      *(char*)pos = (char)c;
+   }
+   return 1;
+}
+#endif
+
+size_t EncodePrintableString(void* dest, const char* src, size_t srcLen, EncodeWChar encodeWChar, void* encState) {
+   assert(src || srcLen == 0);
+
+   size_t srcPos = 0;
+   size_t destSize = 0;
+   bool wasReplaced = false;
+
+#ifdef HAVE_LIBNCURSESW
```

Comment:
Any chance we could split this into two smaller blocks? Currently there is much code duplication.

Reply:
I wished to do that, but the problem is on the data type. One is `char` and the other is `wchar_t`. How do I resolve this? (Introduce something like a `TCHAR` type in Windows programming?)